hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1780,6 +1797,35 @@ int SaraNcpClient::checkSimReadiness(bool checkForRfReset) {
     return r;
 }
 
+int SaraNcpClient::getPowerSavingValue() {
+    auto respUpsv = parser_.sendCommand("AT+UPSV?");
+    int upsvVal = -1;
+    // +UPSV: 1,2000,1
+    auto upsvValCnt = respUpsv.scanf("+UPSV: %d,%*d,%*d", &upsvVal);
+    CHECK_PARSER_OK(respUpsv.readResult());
+    CHECK_TRUE(upsvValCnt == 1, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
+
+    return upsvVal;
+}
+
+int SaraNcpClient::setPowerSavingValue(CellularPowerSavingValue upsv, bool check) {
+    if (check) {
+        if ((int)upsv == CHECK(getPowerSavingValue())) {
+            // Already in required state
+            return SYSTEM_ERROR_NONE;
+        }
+    }
+
+    int r = SYSTEM_ERROR_UNKNOWN;
+
+    r = parser_.execCommand(1000, "AT+UPSV=%d",(int)upsv);
```

Comment:
> How did you observe the CTS1 pin when in power save mode? Did you do testing on an EVK? I think the only board of ours that I might be able to probe is an esomx...

I soldered some tiny tiny wires to the bottom of a Boron :D

> This would be muxer_.setKeepAlivePeriod(0) right? I think it is probably worth doing. Its not clear to me that muxer_.suspendChannel() will stop processTimeouts() from being called and sending a keep alive message.

suspendChannel does stop all keepAlives, and sleepNoPPPWrite_ also ensures they are blocked.  I've tried to keep the changing of keepalives in to places, init and on state change.  That way they are not called repeatedly, and ensures they are called when needed and only when needed.

> We are relying on the default Timeout value of 9.2s for the module to sleep right?

Correct.  We talked about decreasing it, but it seems to work well enough at the default value.